### DISCUSSION TITLE
superfluous response.WriteHeader call from github.com/go-kit/kit/transport/http.DefaultErrorEncode

### DISCUSSION THREAD

**Discussion by nirdothan-zz** [go-kit/kit#1154] [Q&A]
running github.com/go-kit/kit v0.10.0 I keep getting repeated log messages in some production environments (not all) 

`http superfluous response.WriteHeader call from github.com/go-kit/kit/transport/http.DefaultErrorEncoder (server.go:209)`
it also seems to eat up memory until the pod is OOM killed.
I cannot reproduce it on my laptop.
I'd appreciate any hint to tell me where to start looking.

**Comment by peterbourgon** [comment]
It looks like something is calling WriteHeader as part of a response to a client, and then invoking the DefaultErrorEncoder, which is also calling WriteHeader. Either you have to avoid calling WriteHeader if you're also going to encode an error, or you need to replace the DefaultErrorEncoder with one that doesn't call WriteHeader.

This probably isn't causing your OOMs.
